pytket/tests/backend_test.py

```diff
@@ -69,7 +69,7 @@ def input_str_invalid(st: str) -> bool:
 
```

Comment:
Can we add a test that the simulator does throw an error as expected when measurements are included?

Reply:
I also implemented a simple Catch2 exception matcher to work with the `REQUIRE_THROWS_MATCHES` macro. I think that macro is potentially useful for many cases. However, I'm not sure if we already have a better way to do this.